WordPress/src/main/java/org/wordpress/android/ui/sitecreation/domains/SiteCreationDomainsViewModel.kt

```diff
@@ -103,6 +112,22 @@ class SiteCreationDomainsViewModel @Inject constructor(
         isStarted = true
         tracker.trackDomainsAccessed()
         resetUiState()
+        if (purchasingFeatureConfig.isEnabledOrManuallyOverridden()) fetchAndCacheProducts()
+    }
+
+    private fun fetchAndCacheProducts() {
+        launch {
+            val result = productsStore.fetchProducts(TYPE_DOMAINS_PRODUCT)
+            when {
+                result.isError -> {
+                    AppLog.e(AppLog.T.DOMAIN_REGISTRATION, "Error while fetching domain products: ${result.error}")
```

Comment:
I wasn't able to trigger this error flow. Would this be reflected on the UI?

Reply:
> I wasn't able to trigger this error flow. Would this be reflected on the UI?

Thank you for this question @antonis 🙇🏻 
In my opinion we're safe to ignore this on the UI, like it's done in the  the current `Menu` → `Domains` feature, because the worst it can happen is that users will see the listing price in the list, and when they go to the yet-to-be-implemented [checkout step](https://github.com/wordpress-mobile/WordPress-Android/issues/17905) I'm expecting the discount to be applied correctly[^note] (internal ref: pc8HXX-ZF-p2).

[^note]: My understanding is that in the communication with the api we don't send data received from `Products`.

Reply:
Sounds good. Thank you for following up on this Ovi 🙇